### PR TITLE
Add support for staying logged with phone auth & basic logout functionality

### LINKED ISSUE #46: Cache verified phone numbers
Users should only need to verify once every while (not every time)

### PR DESCRIPTION
**Pull request by wangannie**
## Summary
- Resolved #46 with the ability for users to stay logged in after authenticating for the first time by storing a user token upon successful phone auth login to AsyncStorage
  - Note: I'm storing what is returned from `credential.providerId` from Firebase, which is literally `'phone'`. We should probably store the user's `uid` later down the road.
  - `AuthLoadingScreen` (shoutout to [DC Central Kitchen](https://github.com/calblueprint/dccentralkitchen/blob/441b6e1493aa683272e9a104c9b8c0917fc23752/screens/auth/AuthLoadingScreen.js)) checks if the token is set in AsyncStorage and determines whether to navigate to `Login` or `Root`.
- Added ability to log out (I just made it the button on the top right on the home screen) which clears the token from AsyncStorage and navigates back to `AuthLoadingScreen`

## Test Plan
**Staying Logged In**
1. Log in using phone auth
2. Close Expo
3. Open the app again
4. You should be taken straight to the map screen
 
**Log out**
1. On the map screen, click the exit button on the top right to log out
2. You should be taken back to LoginScreen
3. If you close and open expo again, you should still be taken to LoginScreen.

### Next Steps
- Set `userToken` to be something user-specific
- Get rid of the obvious transition between `AuthLoadingScreen` and the next screens
- Clean up logout function

### Relevant Links
https://github.com/calblueprint/dccentralkitchen/blob/441b6e1493aa683272e9a104c9b8c0917fc23752/screens/auth/AuthLoadingScreen.js

(dcck 4 lyfe)
### Online Sources
n/a

### Related PRs
n/a

## Screenshots and Tests
Loom is not cooperating so I can't get a screen recording going unfortunately.

CC: @mylesdomingo

### CODE REVIEW

File: src/navigation/TabNavigator.tsx [hunk 1]
```diff
@@ -27,8 +28,11 @@ export default function TabNavigator() {
           tabBarIcon: ({ color }) => <Icon name="home" color={color} />,
           headerRight: () => (
             <Icon
-              name="info-circle"
-              onPress={() => navigation.navigate('Modal')}
+              name="sign-out"
+              onPress={async () => {
+                await AsyncStorage.removeItem('userToken');
```
Comment (wangannie) [inline src/navigation/TabNavigator.tsx:33]: Honestly, I'm kind of questioning whether or not AsyncStorage is the best thing to use for this. Open to suggestions!

**Comment by mylesdomingo** [comment]
You make my PRs look to shame. Good work.

File: src/navigation/TabNavigator.tsx (hunk 1 above)
Comment (kennethlien) [reply src/navigation/TabNavigator.tsx:33]: Low priority, if AsyncStorage works I think we should stick with it (although I do get lots of warnings when I start up the app.

**Review by kennethlien — APPROVED** [review]
I have disagreements with how ViewContainer takes in props, but we can talk about best practice for this later.

File: package.json [hunk 2]
```diff
@@ -27,8 +27,10 @@
   },
   "dependencies": {
     "@expo/vector-icons": "^12.0.0",
+    "@react-native-community/async-storage": "^1.12.1",
```
Comment (mylesdomingo) [inline package.json:30]: I think there might be an issue with this package along with expo.
https://stackoverflow.com/questions/56029007/nativemodule-asyncstorage-is-null-with-rnc-asyncstorage

Getting an error where Async Storage is null.
See attached:

image

**Review by mylesdomingo — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (wangannie) [reply package.json]: Good catch -- updated to https://react-native-async-storage.github.io/async-storage/docs/usage/

**Review by mylesdomingo — APPROVED** [review]

**Pull request merged**